board/opentrons/ot2/rootfs-overlay/usr/bin/ot-status-leds

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+# Monitor system status and update LEDs accordingly
+# D2 ) Ethernet LED  : STATUS_LED_1  => Pin15    => bcm22
+# D3 ) Wifi LED      : STATUS_LED_2  => Pin27    => bcm00
+# D4 ) Heartbeat LED : STATUS_LED_3  => Pin22    => bcm25
+# D5 ) UNDEFINED_LED : STATUS_LED_4  => Pin28    => bcm01
+
+import gpiod
+import enum
+import asyncio
+import logging
+from logging.config import dictConfig
+from typing import Optional, Any
+
+from opentrons.system import nmcli
+from opentrons.hardware_control.types import BoardRevision
+from opentrons.drivers.rpi_drivers.gpio import GPIOCharDev
+
+
+CHIP = gpiod.Chip('gpiochip0')
+
+BLINK_ON_TIME = 0.25    # seconds
+BLINK_OFF_TIME = 2      # seconds
+NW_STATUS_POLL_SEC = 5
+
+dictConfig({
+        'version': 1,
+        'disable_existing_loggers': False,
+        'formatters': {
+            'message_only': {'format': '%(levelname)s - %(message)s'}
+        },
+        'handlers': {
+            'service': {
+                'class': 'systemd.journal.JournalHandler',
+                'level': logging.INFO,
+                'formatter': 'message_only',
+                'SYSLOG_IDENTIFIER': 'ot-status-leds',
+            }
+        },
+        'loggers': {
+            '__main__': {
+                'level': logging.INFO,
+                'handlers': ['service']
+            }
+        },
+    })
+log = logging.getLogger(__name__)
+
+
+class LedLineOffset(int, enum.Enum):
+    ETH = 22
+    WIFI = 0
+    HEARTBEAT = 25
+    UNDEFINED = 1
+
+    def __init__(self, offset):
+        self._line = CHIP.get_line(offset)
+
+    @property
+    def line(self):
+        return self._line
+
+
+def init_leds() -> None:
+    """ Sets LED pins as output. """
+    for offset in list(LedLineOffset):
+        offset.line.request(consumer='ot-status-leds',
+                            type=gpiod.LINE_REQ_DIR_OUT)
+        offset.line.set_value(0)
+
+
+async def blink_heartbeat_led() -> None:
+    """ Keep heartbeat LED blinking indefinitely. """
+    hb_line = LedLineOffset.HEARTBEAT.line
+    while True:
+        hb_line.set_value(1)
+        await asyncio.sleep(BLINK_ON_TIME)
+        hb_line.set_value(0)
+        await asyncio.sleep(BLINK_OFF_TIME)
+
+
+async def update_nw_status_leds() -> None:
+    """ Update WIFI & LinkLocal status LEDs according to NW status.
+
+    Turn the LED ON when the respective network interface is UP.
+    LED OFF when the interface is DOWN.
+    """
+    while True:
+        eth_status = await nmcli.iface_info(nmcli.NETWORK_IFACES.ETH_LL)
+        update_led(eth_status.get('ipAddress'), LedLineOffset.ETH.line)
```

Comment:
This and the usb one probably also want to check what nmcli says the connection status is, to allow for situations where there's a static ip. It should be up/active/connected/whatever it's called AND have an IP.

Reply:
Oh interesting. Although, there's no such thing as a static ip for link local, is there? As for wifi, does the interface store the static ip address or is it the router that assigns it according to the mac address? In which case, if the interface is down, it won't have an ip address.. it'll have to be up to receive the ip address, won't it?

Reply:
> Although, there's no such thing as a static ip for link local, is there?

Support sometimes has to set up weird networking configurations on customer OT-2s, either to conform to network requirements beyond the customer's control, or to work around stubborn technical problems like mDNS not getting through for unknown reasons.

One of Support's well-used scripts permanently sets the OT-2's wired interface to `169.254.123.123`. It's `UtilityProtocols/configure_static_ip.py` in Support's private repo, if you're curious to see it.

Reply:
Thanks for the script @SyntaxColoring . I tested those settings and looks like checking for ipv4 would be enough. nmcli status doesn't return an ip address when the interface is down.